internal/db/repo_editor.go

```diff
@@ -460,7 +460,7 @@ type UploadRepoFileOptions struct {
 
 // isRepositoryGitPath returns true if given path is or resides inside ".git" path of the repository.
 func isRepositoryGitPath(path string) bool {
-	return strings.HasSuffix(path, ".git") || strings.Contains(path, ".git"+string(os.PathSeparator))
+	return strings.HasSuffix(path, ".git") || strings.Contains(path, ".git"+string(os.PathSeparator) || strings.Contains(path, string(os.PathSeparator)+".git.")
```

Comment:
```suggestion
	return strings.HasSuffix(path, ".git") || strings.Contains(path, ".git"+string(os.PathSeparator) || strings.Contains(path, ".git."+string(os.PathSeparator))
```

Shouldn't the `string(os.PathSeparator)` being concatenated in this way?

Reply:
I recommend this modification for the following reasons.

```suggestion
	return strings.HasSuffix(path, ".git") || strings.HasSuffix(path, ".git.") || strings.Contains(path, ".git"+string(os.PathSeparator) || strings.Contains(path, ".git."+string(os.PathSeparator)
```

Whether it contains a os separator or not, the value of `tree_path` will be treated as a folder when `path.Join`.

ref: /gogs/internal/db/repo_editor.go `targetPath := path.Join(dirPath, upload.Name)`